Can't set up dev environment

<!--
  If this issue affects many people in a company/big team, create a post for your company in the following discussion:
  https://github.com/pnpm/pnpm/discussions/3787
  and link the issue in your post.

  This will help us prioritize issues that affect more people.
-->

### pnpm version: 8.6.1

### Code to reproduce the issue:

Clone this repository and run the following commands inside its directory:

```bash
pnpm i
pnpm link --dir ./pnpm/dev -g
pnpm run compile
pd
```

### Expected behavior:

The help text of `pnpm` gets printed, the same as from running regular production `pnpm`

### Actual behavior:

```
Error: Cannot find module '@pnpm/find-workspace-packages'
Require stack:
- /Users/illright/work/pnpm/pnpm/dev/pd.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1070:15)
    at Module._load (node:internal/modules/cjs/loader:923:27)
    at Module.require (node:internal/modules/cjs/loader:1137:19)
    at require (node:internal/modules/helpers:121:18)
    at Object.<anonymous> (/Users/illright/work/pnpm/pnpm/dev/pd.js:5:42)
    at Module._compile (node:internal/modules/cjs/loader:1255:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1309:10)
    at Module.load (node:internal/modules/cjs/loader:1113:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/Users/illright/work/pnpm/pnpm/dev/pd.js' ]
}

Node.js v20.3.0
```

### Additional information:

 - `node -v` prints: `v20.3.0`
 - Windows, macOS, or Linux?: macOS 13.4

--- a/pnpm/dev/pd.js
+++ b/pnpm/dev/pd.js
@@ -2,7 +2,7 @@
 const fs = require('fs')
 const esbuild = require('esbuild')
 const pathLib = require('path')
-const { findWorkspacePackagesNoCheck } = require('@pnpm/find-workspace-packages')
+const { findWorkspacePackagesNoCheck } = require('@pnpm/workspace.find-packages')
 const { findWorkspaceDir } = require('@pnpm/find-workspace-dir')
 
 const pnpmPackageJson = JSON.parse(fs.readFileSync(pathLib.join(__dirname, 'package.json'), 'utf8'))

